src/common/utils/logger_helper.h

```diff
@@ -28,12 +28,41 @@ namespace LoggerHelpers
         return false;
     }
 
+    inline bool create_dir_if_does_not_exist(std::filesystem::path dir)
+    {
+        std::error_code err;
+        auto entry = std::filesystem::directory_entry(dir, err);
+        if (err.value())
+        {
+            Logger::error("Failed to create directory entry. {}", err.message());
+            return false;
+        }
+
+        if (!entry.exists())
+        {
+            Logger::warn("Directory {} does not exist", dir.string());
```

Comment:
`warning` is for non fatal errors, but still errors. 
Since this method is called `create_dir_if_does_not_exist` the fact that the dir is not present is not something that should generate a warning. If you want you can change it to a verbose/trace but it can be removed as well.

Reply:
It makes sense. Anyway, I removed `create_dir_if_does_not_exist`.